tests/array_test.cpp

```diff
@@ -105,9 +105,23 @@ TEST_CASE("operator<=>", "[SWS_CORE], [SWS_CORE_01201]")
 {
     ara::core::Array<int, 3> a{0, 1, 2};
     ara::core::Array<int, 3> b{0, 1, 2};
-    ara::core::Array<int, 2> c{0, 1};
+    ara::core::Array<int, 3> c{0, 1, 1};
 
-    REQUIRE((a<=>b) == 0);
+    //This is a hack becasue Catch2 can't handle <=> properly
+    //It seems that <=> is not yet supported
+    if((a<=>b) == 0)
+    {
+        CHECK(true);
```

Comment:
No, not like that please :)
just:
```suggestion
        CHECK((a<=>b) == 0);
```
CHECK is a powerful macro, we can put any expression that yields bool inside.

Reply:
@slci that is mine happy coding (stupid idea) :D pushed around 11.30pm yesterday ... Mother of all code what a utter s*** is that :D you are 100% right

@pz0 please do as @slci asked

Reply:
it's generating errors during compilation, catch2 not supporting this opeator

Reply:
@slci Unfortunately I was right with this approach. I am unsure how `Catch2` works in their implementation but it does more than just regular `bool` check - and it could not handle `threeway-comparison` aka `spaceship operator` `<=>`. Therefore we need that workaround. 

We can create and issue to check how to overcome that in `catch2`

![image](https://user-images.githubusercontent.com/4910830/85835682-c0bf6400-b795-11ea-95cf-2bd0e2f0b50c.png)

Reply:
🤦‍♂️ Geez, alright then we have to work this around.
"it does more than just regular bool check" - yes it does, it does some magic to pretty print expression in reports. So it's quite obvious it doesn't support spaceship operatore yet.